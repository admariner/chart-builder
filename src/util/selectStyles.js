// @flow
type StyleType = { [string]: (base: Object, state?: Object) => Object }
const sharedObj: StyleType = {
  option: () => ({
    padding: '4px 12px'
  }),
  control: () => ({
    minHeight: 28,
    height: 28,
    fontSize: 13,
    backgroundColor: '#fff',
    margin: '.25rem 0'
  }),
  dropdownIndicator: () => ({
    padding: 0
  }),
  clearIndicator: () => ({
    padding: 0
  }),
  singleValue: () => ({
    paddingRight: 4
  }),
  valueContainer: () => ({
    overflowX: 'hidden',
    padding: '0 8px'
  }),
  menu: () => ({
    marginTop: 0
  }),
  menuPortal: () => ({
    zIndex: 9999
  })
}

const sharedKeys = Object.keys(sharedObj)

const decorate = (target: StyleType) => {
  const combinedKeys = [...new Set([...Object.keys(target), ...sharedKeys])]
  const obj = {}
  combinedKeys.forEach(k => {
    const s = sharedObj[k]
    const t = target[k]
    obj[k] = (base, state) => ({
      ...base,
      ...(s ? s(base, state) : {}),
      ...(t ? t(base, state) : {})
    })
  })
  return obj
}

export default decorate
