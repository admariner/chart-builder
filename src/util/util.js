// @flow

export function parseParams(searchString: String) {
  const query = new URLSearchParams(searchString)
  const obj = {}
  for (let entry of query) {
    obj[entry[0]] = entry[1]
  }
  return obj
}

export function createParams(objOrString: Object | string) {
  if (typeof objOrString === 'string') return new URLSearchParams(objOrString)

  const query = new URLSearchParams()
  for (let key in objOrString) {
    query.set(key, objOrString[key])
  }
  return query
}

export function getDownloadName(baseName: string, extension: string) {
  const dateStr = new Date().toISOString().replace(/[.:]/g, '-')
  return `${baseName}-${dateStr}.${extension}`
}

export function encodeFieldName(name: string) {
  return name.replace(/([.\[\]])/g, '\\$1')
}
