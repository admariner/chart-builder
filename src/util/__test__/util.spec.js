import { parseParams, createParams, getDownloadName } from '../util'

describe('parseParams', () => {
  it('works', () => {
    expect(parseParams('a=1&b=2')).toMatchSnapshot()
  })
})

describe('createParams', () => {
  it('works with string', () => {
    expect(new Map(createParams('a=1&b=2'))).toMatchSnapshot()
  })

  it('works with object', () => {
    expect(new Map(createParams({ a: 1, b: 2 }))).toMatchSnapshot()
  })
})

describe('getDownloadName', () => {
  it('works', () => {
    const RealDate = Date

    global.Date = class extends RealDate {
      constructor() {
        return new RealDate('2018-04-01T18:00:00.000Z')
      }
    }

    expect(getDownloadName('base-name', 'png')).toMatchSnapshot()
  })
})
