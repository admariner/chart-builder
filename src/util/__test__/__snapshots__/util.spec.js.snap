// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createParams works with object 1`] = `
Map {
  "a" => "1",
  "b" => "2",
}
`;

exports[`createParams works with string 1`] = `
Map {
  "a" => "1",
  "b" => "2",
}
`;

exports[`getDownloadName works 1`] = `"base-name-2018-04-01T18-00-00-000Z.png"`;

exports[`parseParams works 1`] = `
Object {
  "a": "1",
  "b": "2",
}
`;
